docusign/lib/DSAuthCodeGrant.js

```diff
@@ -0,0 +1,263 @@
+    // expires: Moment {_isAMomentObject: true, _isUTC: false, _pf: {…}, _locale: Locale, _d: Tue Jun 26 2018 04:05:37 GMT+0300 (IDT), …}
+    // expiresIn: 28800  // when the token will expire, in seconds, from when the OAuth response is sent by DocuSign
+    // family_name: "LastName" // the user's last name
+    // given_name: "Larry" // the user's first name
+    // name: "Larry LastName"
+    // provider: "docusign"
+    // refreshToken: "eyJ0eXAiOiJ...HB4Q" // Can be used to obtain a new set of access and response tokens.
+    // The lifetime for the refreshToken is typically 30 days
+    // sub: "...5fed18870" // the user's id in guid format
+
+    this.getDefaultAccountInfo(req);
+
+    // If an example was requested, but authentication was needed (and done),
+    // Then do the example's GET now.
+    // Else redirect to home
+    if (req.session.eg) {
+      let eg = req.session.eg;
+      req.session.eg = null;
+      res.redirect(`/${eg}`);
+    } else if (dsConfig.quickstart === 'true') {
+      res.redirect('/eg001');
+    } else { res.redirect('/'); }
+};
+
+/**
+ * Clears the DocuSign authentication session token
+ * https://account-d.docusign.com/oauth/logout
+ * @function
+ */
+DSAuthCodeGrant.prototype.logout = function _logout(req, res) {
+  let logoutCB = encodeURIComponent(res.locals.hostUrl + '/ds/logoutCallback');
+     let oauthServer = dsConfig.dsOauthServer;
+     let client_id = dsConfig.dsClientId;
+     let logoutURL = `${oauthServer}/logout?client_id=${client_id}&redirect_uri=${logoutCB}&response_mode=logout_redirect`
+    ;
+  // console.log (`Redirecting to ${logoutURL}`);
+  // res.redirect(logoutURL);
+
+  // Currently, the OAuth logout API method has a bug: ID-3276
+  // Until the bug is fixed, just do a logout from within this app:
+  this.logoutCallback(req, res);
+};
+
+/**
+ * Clears the user information including the tokens.
+ * @function
+ */
+DSAuthCodeGrant.prototype.logoutCallback = function _logout(req, res) {
+  req.logout(function(err) {
+    if (err) {
+      throw err;
+    }
+  }); // see http://www.passportjs.org/docs/logout/
+  this.internalLogout(req, res);
+    req.flash('info', 'You have logged out.');
+    res.redirect('/');
+};
+
+/**
+ * Clears the object's and session's user information including the tokens.
+ * @function
+ */
+DSAuthCodeGrant.prototype.internalLogout = function _internalLogout(req, res) {
+  this._tokenExpiration = null;
+  req.session.accountId = null;
+  req.session.accountName = null;
+  req.session.basePath = null;
+};
+
+/**
+ * Find the accountId, accountName, and baseUri that will be used.
+ * The dsConfig.targetAccountId may be used to find a specific account (if the user has access to it).
+ * Side effect: store in the session
+ * @function
+ * @param req the request object
+ */
+DSAuthCodeGrant.prototype.getDefaultAccountInfo = function _getDefaultAccountInfo(req) {
+    const targetAccountId = dsConfig.targetAccountId;
```

Comment:
Some of the indents are indented at `2` spaces and some are indented at `4` spaces.